### PR TITLE
Refactor library, adding all features needed for productive usage

### PR DESCRIPTION
**Pull request by davideberlein**
- Extract `BackroundEffectProcessor` class from `BackgroundEffect`:
   - `BackroundEffectProcessor` is responsible for the lifecycle, allowing pausing, switching source, disabling the effect while always keeping one steady output stream.
   - `BackgroundEffect` and its implementation is responsible for the actual background replacement effect
- Update documentation
- Update sample app

See the updated documentation for details

New demo page:
image

### CODE REVIEW

File: lib/effects/BackgroundEffectProcessor.ts [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+     */
+    public get inputStream(): MediaStream {
+        return this._inputStream;
+    }
+
+    /**
+     * Sets / changes the input stream to the given stream. The effect will be applied to this stream and then forwarded to the ouput.
+     * @param stream
+     */
+    public set inputStream(stream: MediaStream) {
```
Comment (enricop89) [inline lib/effects/BackgroundEffectProcessor.ts:216]: I would prefer to keep the getter and setter simple and not to implement the logic into them. So I would suggest adding a function **setInputStream**(stream: MediaStream) which implements the logic and having the setter to just set the inputStream (`this._inputStream = stream;`).

Same thing for outputState and effectEnabled. Wdyt?

**Comment by enricop89** [comment]
Thanks for the PR @davideberlein , I left a comment

File: lib/effects/BackgroundEffectProcessor.ts (hunk 1 above)
Comment (davideberlein) [reply lib/effects/BackgroundEffectProcessor.ts:216]: I agree, I'll make the inputStream readonly then

Comment (davideberlein) [reply lib/effects/BackgroundEffectProcessor.ts:216]: @enricop89 what do you think about the updated version?

Comment (enricop89) [reply lib/effects/BackgroundEffectProcessor.ts:216]: @davideberlein I don't see any new changes. Can you check if you have pushed them please?

Comment (davideberlein) [reply lib/effects/BackgroundEffectProcessor.ts]: when I came back to my computer I saw that the push had failed, sorry.

**Comment by davideberlein** [comment]
@enricop89 have you had any time to look at the changes yet?

**Comment by enricop89** [comment]
LGTM

**Pull request merged**